packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx

```diff
@@ -719,4 +732,50 @@ export class VerticalBarChartBase extends React.Component<IVerticalBarChartProps
         (typeof lineYValue !== 'undefined' ? ` ${lineLegend}, ${lineYValue}.` : '')
     );
   };
+
+  private _renderBarValue(xPoint: number, yPoint: number, barValue: number) {
+    if (this.props.hideValues || this._barWidth < 16) {
+      return null;
+    }
+
+    return (
+      <text
+        x={xPoint + this._barWidth / 2}
+        y={yPoint - 6}
+        textAnchor="middle"
+        className={this._classNames.barValue}
+        aria-hidden={true}
+      >
+        {d3FormatPrefix(barValue < 1000 ? '.2~' : '.1', barValue)(barValue)}
+      </text>
+    );
+  }
+
+  private _getDomainMargins = (containerWidth: number): IMargins => {
```

Comment:
Can you add a doc explaining your computations in diagrams.

Reply:
ok